MoppApp/MoppApp/AddresseeViewController.swift

```diff
@@ -0,0 +1,251 @@
+                }
+                let isAddButtonDisabled = selectedIndexes.contains(row) || isSelected
+                cell.populate(addressee: foundAddressees[row] as! Addressee, index: row, isAddButtonDisabled: isAddButtonDisabled)
+                return cell
+            case .addressees:
+                let cell = tableView.dequeueReusableCell(withType: ContainerAddresseeCell.self, for: indexPath)!
+                cell.delegate = self
+                cell.populate(addressee: selectedAddressees[row] as! Addressee, index: row, showRemoveButton: true)
+                return cell
+        }
+    }
+    
+}
+extension AddresseeViewController : UITableViewDelegate {
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        switch sections[indexPath.section] {
+        case .notifications:
+            break
+        case .search:
+            break
+        case .searchResult:
+            break
+        case .addressees:
+            break
+        }
+    }
+    
+    func tableView(_ tableView: UITableView, viewForHeaderInSection _section: Int) -> UIView? {
+        let section = sections[_section]
+        switch section {
+            case .addressees:
+                if (selectedAddressees.count > 0) {
+                    var title: String!
+                    title = sectionHeaderTitle[section]
+                    
+                    if let header = MoppApp.instance.nibs[.containerElements]?.instantiate(withOwner: self, type: ContainerTableViewHeaderView.self) {
+                        
+                        header.delegate = self
+                        header.populate(
+                            withTitle: title,
+                            showAddButton: false)
+                        return header
+                    }
+                }
+                break
+            default:
+                break
+            
+        }
+        return nil
+        
+    }
+    
+    func tableView(_ tableView: UITableView, heightForHeaderInSection _section: Int) -> CGFloat {
+        let section = sections[_section]
+        switch section {
+            case .addressees:
+                if (selectedAddressees.count > 0) {
+                    return ContainerTableViewHeaderView.height
+                } else {
+                    return 0
+            }
+            default:
+                break
+        }
+        return 0
+    }
+    
+}
+
+extension AddresseeViewController : ContainerTableViewHeaderDelegate {
+    func didTapContainerHeaderButton() {
+        NotificationCenter.default.post(
+            name: .startImportingFilesWithDocumentPickerNotificationName,
+            object: nil,
+            userInfo: [kKeyFileImportIntent: MoppApp.FileImportIntent.addToContainer, kKeyContainerType: LandingViewController.shared.containerType])
+    }
+}
+
+extension AddresseeViewController : ContainerAddresseeCellDelegate {
+    func removeAddressee(index: Int) {
```

Comment:
The logic in this method requires explanation with the comment(s) or maybe restructuring in a more understandable way. 
Also `oldSelectedAddressees` should be renamed to something more understandable, currently it is easy to confuse this variable as an array type.